src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128_1.cs

```diff
@@ -261,6 +310,50 @@ public T ToScalar()
             return Unsafe.As<Vector128<T>, T>(ref Unsafe.AsRef(in this));
         }
 
+        /// <summary>Converts the current instance to an equivalent string representation.</summary>
+        /// <returns>An equivalent string representation of the current instance.</returns>
+        /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        public override string ToString()
+        {
+            return ToString("G");
+        }
+
+        /// <summary>Converts the current instance to an equivalent string representation using the specified format.</summary>
+        /// <param name="format">The format specifier used to format the individual elements of the current instance.</param>
+        /// <returns>An equivalent string representation of the current instance.</returns>
+        /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        public string ToString(string format)
+        {
+            return ToString("G", CultureInfo.CurrentCulture);
+        }
+
+        /// <summary>Converts the current instance to an equivalent string representation using the specified format.</summary>
+        /// <param name="format">The format specifier used to format the individual elements of the current instance.</param>
+        /// <param name="formatProvider">The format provider used to format the individual elements of the current instance.</param>
+        /// <returns>An equivalent string representation of the current instance.</returns>
+        /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        public string ToString(string format, IFormatProvider formatProvider)
+        {
+            ThrowIfUnsupportedType();
+
+            string separator = NumberFormatInfo.GetInstance(formatProvider).NumberGroupSeparator;
+            int lastElement = Count - 1;
+
+            var sb = new StringBuilder();
```

Comment:
I'm not sure how fast we need this to be, but should we use StringBuilderCache here?

Reply:
Yes, if we have an existing cache we use elsewhere. I just followed what `S.N.Vector<T>` was doing for the initial commit here.

Reply:
There are numerous examples, just search for `StringBuilderCache`. Here's an easy example: https://github.com/dotnet/coreclr/blob/3b807944d8822b44eb5085d6b95b130b4a91808f/src/System.Private.CoreLib/shared/System/Collections/Generic/KeyValuePair.cs#L24-L41

> I just followed what S.N.Vector<T> was doing for the initial commit here.

It makes sense that `Vector<T>` wasn't using it because it is only available in `System.Private.CoreLib`, and `Vector<T>` didn't initially live in corelib.

Reply:
Fixed.